### ISSUE TITLE
Problem installing via pip

### ISSUE THREAD

**Issue by Kycilak** [gustavochm/sgtpy#6]
Dear Gustavo,
I have a problem with installing sgtpy:

```
pip3 install sgtpy
Defaulting to user installation because normal site-packages is not writeable
Collecting sgtpy
  Using cached sgtpy-0.0.15.tar.gz (165 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      running egg_info
      creating /tmp/pip-pip-egg-info-9vochw75/sgtpy.egg-info
      writing /tmp/pip-pip-egg-info-9vochw75/sgtpy.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-pip-egg-info-9vochw75/sgtpy.egg-info/dependency_links.txt
      writing requirements to /tmp/pip-pip-egg-info-9vochw75/sgtpy.egg-info/requires.txt
      writing top-level names to /tmp/pip-pip-egg-info-9vochw75/sgtpy.egg-info/top_level.txt
      writing manifest file '/tmp/pip-pip-egg-info-9vochw75/sgtpy.egg-info/SOURCES.txt'
      error: package directory 'sgtpy' does not exist
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

Thank you for your help.

**Comment by gustavochm** [comment]
Dear Kycilak,

Thanks for your interest in sgtpy! 
sgtpy uses cython for some parts of the code, hence it requires a c compiler. For Linux/Mac OS, getting a compiler shouldn't be a problem. 
For Windows, it could be a few extra steps to get the compiler running and for that reason, I uploaded precompiled files to pypi for windows for python 3.6, 3.7, 3.8 and 3.9. Sadly, the last time that I tried there were some compatibility issues with python 3.10.

Could you please let me know which python version and which operative system are you using? 

Gustavo

**Comment by Kycilak** [comment]
I use Linux Mint 21, kernel 5.17.0
I have gcc (and g++) 11.2.0 and Python 3.10.6 installed.
Thank you for swift response.

**Comment by Kycilak** [comment]
I'll try downgrading Python to 3.9 and report back.

**Comment by Kycilak** [comment]
I installed python 3.9.15, installed dependencies for sgtpy (numpy, scipy, cython, pandas, numba) and tried to install sgtpy again, yet I have got the same error:
```
$ python3.9 -m pip -V
pip 22.0.2 from /usr/lib/python3/dist-packages/pip (python 3.9)
$ python3.9 -m pip install sgtpy
Defaulting to user installation because normal site-packages is not writeable
Collecting sgtpy
  Using cached sgtpy-0.0.15.tar.gz (165 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      running egg_info
      creating /tmp/pip-pip-egg-info-7n_y_nd0/sgtpy.egg-info
      writing /tmp/pip-pip-egg-info-7n_y_nd0/sgtpy.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-pip-egg-info-7n_y_nd0/sgtpy.egg-info/dependency_links.txt
      writing requirements to /tmp/pip-pip-egg-info-7n_y_nd0/sgtpy.egg-info/requires.txt
      writing top-level names to /tmp/pip-pip-egg-info-7n_y_nd0/sgtpy.egg-info/top_level.txt
      writing manifest file '/tmp/pip-pip-egg-info-7n_y_nd0/sgtpy.egg-info/SOURCES.txt'
      error: package directory 'sgtpy' does not exist
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
$
```
I have noticed you wrote Python3.10 is problematic for Windows so maybe it isn't surprising that is not the problem but I have no idea why it does not work.

**Comment by gustavochm** [comment]
Hi Kycilak,

To be honest I am not really sure what may be the problem :/ Just by googling, they point [this error can be related to setuptools](https://www.partitionwizard.com/partitionmanager/python-setup-py-egg-info-failed-with-error-code-1.html). Can you check if setuptools is up to date?

An alternative option would be to avoid installing through pip and instead download sgtpy directly from github and then install it by running on that downloaded directory `python setup.py install`.

Apologies that I cannot be of more help.

Gustavo

**Comment by Kycilak** [comment]
Hello Gustavo,
I've got it running, I am just not entirely sure how.

The problem with installation seemed to be that I did not have pip user installation directory in PATH so SGTPy could not import it?
I've installed cython through ```apt install``` because that installed it to dir in PATH (I did not know yet about PATH) and that allowed me to ```sudo python3 setup.py install```. Installation through pip did not work still.
I then ran into another problem where math.py could not import coloc_cy when I tried to import from sgtpy. I tried to import math alone and that worked and from that point everything works.
I have yet to discover whether it will work after reboot.
I do not know much about Python or Linux yet and I am sorry if I made a trivial mistake.

Thank you so much for your support, Gustavo. The package itself is wonderful, continue doing great work.
Have a nice day
